tokenizers/src/normalizers/mod.rs

```diff
@@ -103,4 +246,36 @@ mod tests {
             NormalizerWrapper::Prepend(_)
         ));
     }
+
+    #[test]
+    fn normalizer_serialization() {
+        let json = r#"{"type":"Sequence","normalizers":[]}"#;
+        assert!(serde_json::from_str::<NormalizerWrapper>(json).is_ok());
+        let json = r#"{"type":"Sequence","normalizers":[{}]}"#;
+        let parse = serde_json::from_str::<NormalizerWrapper>(json);
+        match parse {
+            Err(err) => assert_eq!(
+                format!("{err}"),
+                "data did not match any variant of untagged enum NormalizerUntagged"
+            ),
+            _ => panic!("Expected error"),
+        }
+
+        let json = r#"{"replacement":"▁","prepend_scheme":"always"}"#;
+        let parse = serde_json::from_str::<NormalizerWrapper>(json);
+        match parse {
+            Err(err) => assert_eq!(
+                format!("{err}"),
+                "data did not match any variant of untagged enum NormalizerUntagged"
+            ),
+            _ => panic!("Expected error"),
+        }
+
+        let json = r#"{"type":"Sequence","prepend_scheme":"always"}"#;
+        let parse = serde_json::from_str::<NormalizerWrapper>(json);
+        match parse {
+            Err(err) => assert_eq!(format!("{err}"), "missing field `normalizers`"),
+            _ => panic!("Expected error"),
+        }
```

Comment:
let's add a precompiles test as you mentioned it's a bit particular

Reply:
The issue is that valid charmaps are extremely long, polluting code with several kb of data.

`tests/serialization.rs` already contains the test that caught this.